planning/behavior_path_planner/include/behavior_path_planner/planner_manager.hpp

```diff
@@ -0,0 +1,257 @@
+using tier4_autoware_utils::generateUUID;
+using tier4_autoware_utils::StopWatch;
+using unique_identifier_msgs::msg::UUID;
+using ModuleID = std::pair<std::shared_ptr<SceneModuleManagerInterface>, UUID>;
+
+struct SceneModuleStatus
+{
+  explicit SceneModuleStatus(const std::string & n) : module_name(n) {}
+
+  std::string module_name;
+
+  bool is_execution_ready{false};
+  bool is_waiting_approval{false};
+
+  ModuleStatus status{ModuleStatus::SUCCESS};
+};
+
+class PlannerManager
+{
+public:
+  PlannerManager(rclcpp::Node & node, const bool verbose);
+
+  BehaviorModuleOutput run(const std::shared_ptr<PlannerData> & data);
+
+  void registerSceneModuleManager(
+    const std::shared_ptr<SceneModuleManagerInterface> & scene_module_manager_ptr)
+  {
+    RCLCPP_INFO(logger_, "register %s module", scene_module_manager_ptr->getModuleName().c_str());
+    scene_manager_ptrs_.push_back(scene_module_manager_ptr);
+    processing_time_.emplace(scene_module_manager_ptr->getModuleName(), 0.0);
+  }
+
+  void updateModuleParams(const std::vector<rclcpp::Parameter> & parameters)
+  {
+    std::for_each(
+      scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(),
+      [&parameters](const auto & m) { m->updateModuleParams(parameters); });
+  }
+
+  void reset()
+  {
+    approved_modules_.clear();
+    candidate_module_id_ = boost::none;
+    root_lanelet_ = boost::none;
+    std::for_each(
+      scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(), [](const auto & m) { m->reset(); });
+    resetProcessingTime();
+  }
+
+  void publishDebugMarker()
+  {
+    std::for_each(scene_manager_ptrs_.begin(), scene_manager_ptrs_.end(), [](const auto & m) {
+      m->publishDebugMarker();
+    });
+  }
+
+  std::vector<std::shared_ptr<SceneModuleManagerInterface>> getSceneModuleManagers() const
+  {
+    return scene_manager_ptrs_;
+  }
+
+  std::vector<std::shared_ptr<SceneModuleStatus>> getSceneModuleStatus() const
+  {
+    std::vector<std::shared_ptr<SceneModuleStatus>> ret;
+
+    for (const auto & m : approved_modules_) {
+      const auto & manager = m.first;
+      const auto & uuid = m.second;
+      auto s = std::make_shared<SceneModuleStatus>(manager->getModuleName());
+      s->is_waiting_approval = manager->isWaitingApproval(uuid);
+      s->status = manager->getCurrentStatus(uuid);
+      ret.push_back(s);
+    }
+
+    if (!!candidate_module_id_) {
+      const auto & manager = candidate_module_id_.get().first;
+      const auto & uuid = candidate_module_id_.get().second;
+      auto s = std::make_shared<SceneModuleStatus>(manager->getModuleName());
+      s->is_waiting_approval = manager->isWaitingApproval(uuid);
+      s->status = manager->getCurrentStatus(uuid);
+      ret.push_back(s);
+    }
+
+    return ret;
+  }
+
+  void resetRootLanelet(const std::shared_ptr<PlannerData> & data);
+
+  void print() const;
+
+private:
+  BehaviorModuleOutput run(
+    const ModuleID & id, const BehaviorModuleOutput & previous_module_output) const
```

Comment:
Does using `ModuleID` as one of the arguments mean that planner manager manages UUID?
If it is true, I think planner manager should manage one of the _status_ from scene modules, not UUID.

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2919/commits/00be57682e73ac2a98333bbba6f56ab681523ea8.